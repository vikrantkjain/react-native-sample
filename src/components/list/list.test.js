import React from 'react';
import List from '.';
import renderer from 'react-test-renderer';
import {View} from 'react-native';
describe('Test List component', () => {
  const tree = renderer.create(
    <List
      data={[1, 2]}
      renderItem={({item}) => {
        return <View />;
      }}
    />,
  );
  it('Renders correctly', () => {
    expect(tree).toBeDefined();
  });
  it('Render successfully', () => {
    const treeToJSON = tree.toJSON();
    expect(treeToJSON).toMatchSnapshot();
  });
});
