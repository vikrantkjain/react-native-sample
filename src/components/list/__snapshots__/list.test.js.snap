// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test List component Render successfully 1`] = `
<View
  style={
    {
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <RCTScrollView
    applyWindowCorrection={[Function]}
    canChangeSize={true}
    contentContainerStyle={
      {
        "backgroundColor": undefined,
        "minHeight": 1,
        "minWidth": 1,
        "paddingLeft": 0,
        "paddingRight": 0,
      }
    }
    contentHeight={100}
    contentWidth={400}
    data={
      [
        1,
        2,
      ]
    }
    dataProvider={
      DataProvider {
        "_data": [
          1,
          2,
        ],
        "_firstIndexToProcess": 0,
        "_hasStableIds": true,
        "_requiresDataChangeHandling": false,
        "_size": 2,
        "getStableId": [Function],
        "rowHasChanged": [Function],
      }
    }
    disableRecycling={false}
    estimatedItemSize={50}
    extendedState={{}}
    externalScrollView={[Function]}
    finalRenderAheadOffset={250}
    forceNonDeterministicRendering={true}
    horizontal={false}
    initialOffset={0}
    initialRenderIndex={0}
    isHorizontal={false}
    keyExtractor={[Function]}
    layoutProvider={
      GridLayoutProviderWithProps {
        "_acceptableRelayoutDelta": 1,
        "_getHeightOrWidth": [Function],
        "_getLayoutTypeForIndex": [Function],
        "_getSpan": [Function],
        "_hasExpired": false,
        "_isHorizontal": false,
        "_lastLayoutManager": GridLayoutManager {
          "_acceptableRelayoutDelta": 1,
          "_getSpan": [Function],
          "_isGridHorizontal": false,
          "_isHorizontal": false,
          "_layoutProvider": [Circular],
          "_layouts": [
            {
              "height": 50,
              "type": 0,
              "width": 400,
              "x": 0,
              "y": 0,
            },
            {
              "height": 50,
              "type": 0,
              "width": 400,
              "x": 0,
              "y": 50,
            },
          ],
          "_maxSpan": 1,
          "_renderWindowSize": {
            "height": 900,
            "width": 400,
          },
          "_totalHeight": 100,
          "_totalWidth": 400,
          "_window": {
            "height": 900,
            "width": 400,
          },
        },
        "_maxSpan": 1,
        "_renderWindowSize": {
          "height": 900,
          "width": 400,
        },
        "_setLayoutForType": [Function],
        "_tempDim": {
          "height": 0,
          "width": 0,
        },
        "averageWindow": AverageWindow {
          "currentAverage": 50,
          "currentCount": 1,
          "inputValues": [
            50,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
            ,
          ],
          "nextIndex": 1,
        },
        "defaultEstimatedItemSize": 100,
        "layoutObject": {
          "size": undefined,
          "span": undefined,
        },
        "props": {
          "data": [
            1,
            2,
          ],
          "estimatedItemSize": 50,
          "keyExtractor": [Function],
          "numColumns": 1,
          "renderItem": [Function],
        },
        "renderWindowInsets": {
          "height": 0,
          "width": 0,
        },
        "shouldRefreshWithAnchoring": true,
      }
    }
    maxRenderAhead={750}
    numColumns={1}
    onEndReached={[Function]}
    onEndReachedThreshold={0}
    onEndReachedThresholdRelative={0}
    onItemLayout={[Function]}
    onLayout={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onSizeChanged={[Function]}
    onVisibleIndicesChanged={[Function]}
    removeClippedSubviews={false}
    renderAheadOffset={0}
    renderAheadStep={250}
    renderContentContainer={[Function]}
    renderItem={[Function]}
    renderItemContainer={[Function]}
    rowRenderer={[Function]}
    scrollEventThrottle={16}
    scrollThrottle={16}
    scrollViewProps={
      {
        "contentContainerStyle": {
          "backgroundColor": undefined,
          "minHeight": 1,
          "minWidth": 1,
          "paddingLeft": 0,
          "paddingRight": 0,
        },
        "onLayout": [Function],
        "onScrollBeginDrag": [Function],
        "refreshControl": undefined,
        "style": {
          "minHeight": 1,
          "minWidth": 1,
        },
      }
    }
    style={
      {
        "minHeight": 1,
        "minWidth": 1,
      }
    }
    suppressBoundedSizeException={true}
    windowCorrectionConfig={
      {
        "applyToInitialOffset": true,
        "applyToItemScroll": true,
        "value": {
          "endCorrection": 0,
          "startCorrection": 0,
          "windowShift": -0,
        },
      }
    }
  >
    <View>
      <View
        style={
          {
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            {
              "paddingLeft": 0,
              "paddingTop": 0,
            }
          }
        />
        <View
          style={
            [
              undefined,
              undefined,
            ]
          }
        />
        <AutoLayoutView
          enableInstrumentation={false}
          horizontal={false}
          onBlankAreaEvent={[Function]}
          onLayout={[Function]}
          renderAheadOffset={0}
          scrollOffset={0}
          style={
            {
              "height": 100,
              "width": 400,
            }
          }
          windowSize={900}
        >
          <CellContainer
            index={0}
            onLayout={[Function]}
            style={
              {
                "alignItems": "stretch",
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": 400,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <View />
            </View>
          </CellContainer>
          <CellContainer
            index={1}
            onLayout={[Function]}
            style={
              {
                "alignItems": "stretch",
                "flexDirection": "column",
                "left": 0,
                "position": "absolute",
                "top": 50,
                "width": 400,
              }
            }
          >
            <View
              style={
                {
                  "flexDirection": "column",
                }
              }
            >
              <View />
            </View>
          </CellContainer>
        </AutoLayoutView>
        <CellContainer
          index={-1}
          style={
            [
              undefined,
              undefined,
            ]
          }
        />
        <View
          style={
            {
              "paddingBottom": 0,
              "paddingRight": 0,
            }
          }
        />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
